src/mat.rs

```diff
@@ -10,11 +10,12 @@ use std::path::Path;
 use std::slice;
 use std::ops::{BitAnd, BitOr, BitXor, Not};
 
+/// The class `CMat` is used as a pointer to represent the Mat opencv structure
 #[derive(Clone, Copy, Debug)]
-pub(crate) enum CMat {}
+pub enum CMat {}
 
 impl CMat {
-    pub fn new() -> *mut CMat {
+    pub(crate) fn new() -> *mut CMat {
```

Comment:
I did this to ensure that you don't create `CMat`, but rather get it from a `Mat`. Debatable decision.

Reply:
I completely agree on that